packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -166,6 +166,14 @@ export default createRule<Options, MessageId>({
      * if the type of the node is always true or always false, it's not necessary.
      */
     function checkNode(node: TSESTree.Expression): void {
+      // Check if the node is Unary Negation expression and handle it
+      if (
+        node.type === AST_NODE_TYPES.UnaryExpression &&
+        node.operator === '!'
+      ) {
+        return checkIfUnaryNegationExpressionIsNecessaryConditional(node);
```

Comment:
perhaps this should just call `checkNode(node.argument)`?

Reply:
Thank you for your review @bradzacher. If checkNode is called here, the following example reports alwaysTruthy, but I believe that the expectation is alwaysFalsy. 

```js
const a = true;
if (!a) {
}

assert.strictEqual(received, expected)

Expected value to strictly be equal to:
  "alwaysFalsy"
Received:
   "alwaysTruthy"
``` 

Have added the new function to handle this. Please suggest if there is better way to do this.